### DISCUSSION TITLE
updated to php 8, composer still installs mpd 6.

### DISCUSSION THREAD

**Discussion by sasori7** [mpdf/mpdf#1599] [Q&A]
I've been wrestling with this for an entire day.
I followed the instructions for installation and noticed that composer installed only mpdf 6 and the example wouldn't work. So, I changed my php to 8.01 (having to upgrade my mamp pro) and still can't get mpdf 7.
`C:\MAMP\htdocs\administration>composer require mpdf/mpdf
Using version ^6.1 for mpdf/mpdf
./composer.json has been created
Running composer update mpdf/mpdf
Loading composer repositories with package information
Updating dependencies
Lock file operations: 2 installs, 0 updates, 0 removals

    Locking mpdf/mpdf (v6.1.3)
    Locking setasign/fpdi (1.6.2)
    Writing lock file
    Installing dependencies from lock file (including require-dev)
    Package operations: 2 installs, 0 updates, 0 removals
    Installing setasign/fpdi (1.6.2): Extracting archive
    Installing mpdf/mpdf (v6.1.3): Extracting archive
    3 package suggestions were added by new dependencies, use composer suggest to see details.
    Generating autoload files

I deleted everything and tried again; same thing. When I run the first example (and the 'try' version) provided, I get:

    [16-Jan-2022 11:52:53 Etc/GMT-9] PHP Fatal error: Uncaught Error: Call to undefined function Mpdf\Mpdf() in C:\MAMP\htdocs\administration\hello.php:11`
    What else must I do?

**Comment by finwe** [comment]
I guees you changed PHP version for server, but not for console. What does `composer --version -vvv` output?

**Reply by finwe** [reply]
Or try to run `composer why-not mpdf/mpdf 8.0.1`

**Reply by sasori7** [reply]
Composer version 2.2.4 2022-01-08 12:30:42
I had deleted everything and reinstalled composer, via the exe, and basically started over... I thought.

**Reply by finwe** [reply]
That command will also print PHP version on the first line of output, that is the sole reason for the -vvv flag

**Reply by sasori7** [reply]
__root__   -       requires          mpdf/mpdf (^6.1)
mpdf/mpdf  v8.0.1  requires          php (^5.6 || ~7.0.0 || ~7.1.0 || ~7.2.0 || ~7.3.0 but 8.0.1 is installed)
mpdf/mpdf  v8.0.1  requires          ext-gd (* but it is missing)
mpdf/mpdf  v8.0.1  requires          setasign/fpdi (^2.1)
__root__   -       does not require  setasign/fpdi (but 1.6.2 is installed)

**Reply by finwe** [reply]
there you go, install GD extension.

**Reply by sasori7** [reply]
nifty. I'm on a windows box running mamp and apparently this issue hasn't been entirely addressed by MAMP even on MAC (or so I've read). So... 
If I try running 7.2, which has php_gd2.dll will it work then? I just need to generate kindergarten worksheets to email during lockdowns.

**Reply by finwe** [reply]
PHP 8 is supported since mPDF v8.0.10. With PHP 7.2 you can run mPDF <7.1.7
